### PR TITLE
Intake

### PR DESCRIPTION
**Pull request by greenblitz4590**
<3

### CODE REVIEW

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeRoller.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package edu.greenblitz.tobyDetermined.subsystems.intake;
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeRoller.java:1]: No get functions?
No states?
Can you roll in an object while extender is closed?

**Review by roygold8 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/tobyDetermined/RobotMap.java [hunk 2]
```diff
@@ -204,4 +200,18 @@ public static class Autonomus {
 		}
 		
 	}
+	
+	public static class Intake{
+		public static final int ROLLER_ID = 0;
+		public static final boolean INVERTED = false;
+		public static final double RAMP_RATE = 0.1;
+		public static final double DEFAULT_POWER = 1;
+		public static final int CUR_LIMIT = 40;
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/RobotMap.java:209]: Consider finding a slightly better name.

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeExtender.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	public void retract() {
+		setValue(DoubleSolenoid.Value.kReverse);
+	}
+	
+	public boolean isExtended() {
+		return solenoid.get().equals(DoubleSolenoid.Value.kForward);
+	}
+	
+	public void toggleExtender() {
+		System.out.println(isExtended());
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeExtender.java:37]: Debug prints? Kinda cringe bro

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeRoller.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+		motor.config(
+				new GBSparkMax.SparkMaxConfObject()
+						.withCurrentLimit(RobotMap.Intake.CUR_LIMIT)
+						.withInverted(RobotMap.Intake.INVERTED)
+						.withRampRate(RobotMap.Intake.RAMP_RATE)
+		);
+	}
+	
+	public static IntakeRoller getInstance(){
+		if(instance != null) {instance = new IntakeRoller();}
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeRoller.java:22]: If you're gonna do one-liners, at least remove the curly braces

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeExtender.java (hunk 3 above)
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeExtender.java:37]: romi's fault

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeExtender.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+public class IntakeExtender extends GBSubsystem {
+	private static IntakeExtender instance;
+	private final DoubleSolenoid solenoid;
+	
+	private IntakeExtender(){
+		solenoid = new DoubleSolenoid(RobotMap.Pneumatics.PCM.PCM_ID, RobotMap.Pneumatics.PCM.PCM_TYPE, RobotMap.Intake.Solenoid.FORWARD_PORT, RobotMap.Intake.Solenoid.REVERSE_PORT);
+	}
+	
+	public static IntakeExtender getInstance(){
+		if(instance == null) {instance = new IntakeExtender();}
```
[the remark this answers is not in the document]
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeExtender.java:16]: romi's fault

File: src/main/java/edu/greenblitz/tobyDetermined/commands/intake/Roller/RollerCommand.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package edu.greenblitz.tobyDetermined.commands.intake.Roller;
+
+import edu.greenblitz.tobyDetermined.subsystems.intake.IntakeRoller;
+import edu.greenblitz.utils.GBCommand;
+
+public abstract class RollerCommand extends GBCommand {
+	IntakeRoller roller;
+	RollerCommand(){
+		super();
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/intake/Roller/RollerCommand.java:9]: Not a deal breaker, but just to let you know.
This line is being called by default, you don't have to write this explicit. (In general, when you don't call a super constructor in the first line, super() is called by deafult.)

File: src/main/java/edu/greenblitz/tobyDetermined/commands/intake/extender/ExtenderCommand.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package edu.greenblitz.tobyDetermined.commands.intake.extender;
+
+import edu.greenblitz.tobyDetermined.subsystems.intake.IntakeExtender;
+import edu.greenblitz.utils.GBCommand;
+
+public abstract class ExtenderCommand extends GBCommand {
+	IntakeExtender extender;
+	public ExtenderCommand(){
+		super();
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/intake/extender/ExtenderCommand.java:9]: Again about the default call to super(); (You don't have to edit it)

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeExtender.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/intake/IntakeExtender.java:16]: Raz - But even better, don't do it as a one linner

File: src/main/java/edu/greenblitz/tobyDetermined/commands/intake/Roller/RollerCommand.java (hunk 6 above)
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/commands/intake/Roller/RollerCommand.java:9]: DONE

**Pull request merged**